test/smtp.test.lua

```diff
@@ -146,6 +166,52 @@ test:test("smtp.client", function(test)
                   m.text,
                   "Subject: =%?utf%-8%?b%?YWJjZGVmZ2hpamvRj2xtbm9wcXJzdHV2d3h5eg==%?=", ""))
     test:is(subj, 1, 'subject codes >127')
-end)
 
+    r = client:request(addr, '3xx@tarantool.org',
+                       'receiver@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'MAIL failed: 354', 'Errors 3xx')
+    test:is(r.status, 354, 'Unexpected code')
+
+    r = client:request(addr, '4xx@tarantool.org',
+                       'receiver@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'MAIL failed: 421', 'service unavailable')
+    test:is(r.status, 421, 'Unexpected code')
+
+    r = client:request(addr, '5xx@tarantool.org',
+                       'receiver@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'MAIL failed: 510', 'unexisting recipient')
+    test:is(r.status, 510, 'Unexpected code')
+
+    r = client:request(addr, 'breakconnect@tarantool.org',
+                       'receiver@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'response reading failed', 'unexpected response code')
```

Comment:
Maybe test that we have `-1` is `r.status` here?

Reply:
Added.